### DISCUSSION TITLE
Is there a way to prefetch clientLoaders in remix SPA mode

### DISCUSSION THREAD

**Discussion by obrassard** [remix-run/remix#9934] [Q&A] [answered]
In a [Remix SPA Mode](https://remix.run/docs/en/main/guides/spa-mode) app, let's say I have this route : 

routes/cats.tsx
```tsx
export async function clientLoader() {
    const data = await fetch("https://meowfacts.herokuapp.com/")
                .then((res) => res.json());

    return json(data);
}

export default function CatPage() {
    const catFacts = useLoaderData();
    return (
        <div className="font-sans p-4">
            <h1 className="text-3xl">Cat facts</h1>
            <p>{catFacts.data[0]}</p>
        </div>
    );
}
```

And somewhere I have a `<Link>` to my cats.tsx page
```tsx
<Link to="/cats" prefetch="intent">
     Cat facts
</Link>
```

Is there a way to configure Remix so that, when prefetching the cats.tsx route, it already executes the `clientLoader` (to prefetch the data before navigating to the new route) ? Right now, it seems that the `clientLoader` is not being executed before the actual navigation to the route?

If this is not possible, what is the real `useCase` of prefetching in SPA Mode ?

**Comment by sergiodxa** [comment, marked as answer]
It's not possible, at least right now, to prefetch a clientLoader.

> If this is not possible, what is the real useCase of prefetching in SPA Mode?

It will still prefetch the code for the next route, without that the user will need to wait for the next route code to download so it can call the client loaders.

**Reply by obrassard** [reply]
Thank you for the clarification
